llvm/lib/Transforms/IPO/SampleProfile.cpp

```diff
@@ -439,7 +439,10 @@ struct CandidateComparer {
 
     const FunctionSamples *LCS = LHS.CalleeSamples;
     const FunctionSamples *RCS = RHS.CalleeSamples;
-    assert(LCS && RCS && "Expect non-null FunctionSamples");
+    // In inline replay mode, CalleeSamples may be null and the order doesn't
+    // matter.
+    if (!LCS || !RCS)
+      return false;
```

Comment:
I think we should give a deterministic order when only one side is null. With the current change, "non-null" vs "null" and "null" vs "non-null" both are false..